### ISSUE TITLE
Alignment detection does not factor in how other styling rules change alignment

### ISSUE THREAD

**Issue by lorenzwalthert** [r-lib/styler#776]
Otherwise, text might be outdated and we detect a false positive. Example?
```r
gle(
  1212, 232, f(n= 2),
  1,      2,  "Flar"
)
```
Seems aligned on commas, but then later, a space is added after `n` and the result of styling is 
```r
gle(
  1212, 232, f(n = 2),
  1,      2,  "Flar"
)
```
Which is written to the cache (if the cache is active). However, now, the code is not aligned anymore and should therefore be styled like 
```r
gle(
  1212, 232, f(n = 2),
  1, 2, "Flar"
)
```

**Edit:** Actually not really possible to fix the problem. Doing it early means changes of other styling rules are not considered when determining alignment detection. They also won't be considered when alignment detection is done at the end.
